airflow/contrib/operators/sqoop_operator.py

```diff
@@ -156,9 +176,15 @@ def execute(self, context):
                 input_lines_terminated_by=self.input_lines_terminated_by,
                 input_optionally_enclosed_by=self.input_optionally_enclosed_by,
                 batch=self.batch,
-                relaxed_isolation=self.relaxed_isolation)
+                relaxed_isolation=self.relaxed_isolation,
+                extra_export_options=self.extra_export_options)
         elif self.cmd_type == 'import':
-            if not self.table:
+            # add create hcatalog table to extra import options if option passed
+            # if new params are added to constructor can pass them in here so don't modify sqoop_hook for each param
+            if self.create_hcatalog_table:
+                self.extra_import_options['create-hcatalog-table'] = ''
+
+            if self.table:
```

Comment:
previously was `if not self.table` which is incorrect, it should be `if self.table` then run the import for the table